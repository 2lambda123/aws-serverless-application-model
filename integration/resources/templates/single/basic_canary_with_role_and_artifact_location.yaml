Parameters:
  CanaryName:
    Type: String
    Default: integration-test
    MaxLength: 21
  S3Name:
    Type: String
    Default: sam-integration-test-bucket
  RoleName:
    Type: String
    Default: sam-integration-test-role
Resources:
  # role and role permission created for now so that canary will develop, won't be needed in the future when feature to
  # automatically create role from serverless canary is added
  SamIntegrationTestRole:
    Type: AWS::IAM::Role
    Properties:
      RoleName:
        Fn::Sub: ${RoleName}-${CanaryName}-${AWS::Region}
      Description: CloudWatch Synthetics lambda execution role for running canaries
      AssumeRolePolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
            Action: sts:AssumeRole

  RolePermissions:
    Type: AWS::IAM::Policy
    Properties:
      Roles:
        - Ref: SamIntegrationTestRole
      PolicyName:
        Fn::Sub: ${RoleName}-policy-${CanaryName}-${AWS::Region}
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - s3:PutObject
              - s3:GetBucketLocation
            Resource:
              - Fn::Sub: arn:aws:s3:::${ResultsBucket}/*
          - Effect: Allow
            Action:
              - logs:CreateLogStream
              - logs:PutLogEvents
              - logs:CreateLogGroup
            Resource:
              - Fn::Sub: arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/cwsyn-test-*
          - Effect: Allow
            Action:
              - s3:ListAllMyBuckets
              - xray:PutTraceSegments
            Resource: '*'
          - Effect: Allow
            Resource: '*'
            Action: cloudwatch:PutMetricData
            Condition:
              StringEquals:
                cloudwatch:namespace: CloudWatchSynthetics

  ResultsBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName:
        # have to add the account id to make sure the bucket name is unique and the bucket deploys correctly
        Fn::Sub: ${S3Name}-${CanaryName}
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256

  SyntheticsCanary:
    Type: AWS::Serverless::Canary
    Properties:
      FunctionName:
        Fn::Sub: ${CanaryName}
      Handler: pageLoadBlueprint.handler
      InlineCode: |
        def handler(event, context):
          print("Hello, world!")
      Role:
        Fn::GetAtt:
          - SamIntegrationTestRole
          - Arn
      Runtime: syn-python-selenium-1.0
      ArtifactS3Location:
        Fn::Join:
          - ''
          - - s3://
            - Ref: ResultsBucket
      StartCanaryAfterCreation: true
      Schedule:
        Expression: rate(0 minute)