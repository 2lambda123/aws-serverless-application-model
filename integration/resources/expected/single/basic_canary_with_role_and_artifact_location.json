[
  { "LogicalResourceId":"SyntheticsCanary", "ResourceType":"AWS::Synthetics::Canary" },
  { "LogicalResourceId":"SamIntegrationTestRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"RolePermissions", "ResourceType":"AWS::IAM::Policy" },
  { "LogicalResourceId":"ResultsBucket", "ResourceType":"AWS::S3::Bucket" }
]